invenio_github/tasks.py

```diff
@@ -144,13 +144,15 @@ def process_release(release_id, verify_sender=False):
     except Exception as ex:
         error_handlers = current_github.release_error_handlers
         release.model.status = ReleaseStatus.FAILED
-
         for error_cls, handler in error_handlers + DEFAULT_ERROR_HANDLERS:
             if isinstance(ex, error_cls):
                 handler(release, ex)
                 current_app.logger.exception(
                     u'Error while processing GitHub release')
                 break
+        if error_cls is Exception:
```

Comment:
The `error_cls` variable only exists in the `for ...` loop above. We might need to change the error-handling and retry logic overall here:

- We retry a task, only if it fails for a specific set of exceptions (defined in a `GITHUB_RELEASE_RETRY_EXCEPTIONS` config variable). When that happens, we set the `release.model.error` message to something like "Release failed for an unknown reason; automatic retry scheduled."
- Otherwise, we allow the default handler to handle the error message (and no retry takes place, since e.g. if the `.zenodo.json` metadata is invalid, this is something that a retry won't help with
- After the maximum number of retries, we also allow the exception handler to set the error (if you check in the Celery docs, there is a special `RetryError` exception raised when you call `.retry()` after the maximum number of retries is performed)

Reply:
The ```error_cls``` variable will be the one that matched the exception when the loop ends.